### ISSUE TITLE
Automatically add Wikimedia Commons links

### ISSUE THREAD

**Issue by ddpasa** [openstreetmap/iD#9610]
### Description

Wikimedia Commons is a wiki project for hosting open media (pictures, videos...).  It is increasingly being used to add media files to PoCs in openstreetmaps.  As of April 2023, there are more than 112k uses of this tag in openstreetmaps (and growing fast, it was only 20k in 2020!).

[See taginfo page](https://taginfo.openstreetmap.org/keys/wikimedia_commons#overview)

We make it easier for contributors to add wikimedia tags.  When someone adds a wikipedia tag, ID automatically adds the corresponding wikidata tag.  The same works in the other direction as well, the wikipedia tag is added automaticaly when one adds the wikidata tag in ID.

We can add the wikimedia commons tag automatically at the same time.  Wikimedia commons categories are often available in the wikipedia or wikidata pages.  For example:

wikidata page: https://www.wikidata.org/wiki/Q15533769

![2023-04-30_12-24](https://user-images.githubusercontent.com/112642920/235348117-2a9456ad-028b-4979-970e-9537e05c14f3.png)

![2023-04-30_12-25](https://user-images.githubusercontent.com/112642920/235348116-9dea9df1-92ea-4086-89d8-f486e4457a5e.png)

**Comment by bhousel** [comment]
I think there is huge benefit to linking notable features in OpenStreetMap to a Wikidata identifier, and I've been a supporter of this effort going back years..

But I don't see any additional benefit to adding more tags for more links to things that can just be linked through that Wikidata identifier - and the Wikimedia Commons category is one of these things.  It's just an extra link that needs to be maintained right?

**Comment by ddpasa** [comment]
Unfortunately we have to maintain a wikimedia commons tag in openstreetmaps, we can't just use wikidata.  There are many PoCs that are not prominent enough to have a wikidata entry, but still have images in wikimedia commons.  We can't just add the wikidata and be done with it.

**Comment by bhousel** [comment]
> There are many PoCs that are not prominent enough to have a wikidata entry, but still have images in wikimedia commons. 

can you give an example?

**Comment by ddpasa** [comment]
> > There are many PoCs that are not prominent enough to have a wikidata entry, but still have images in wikimedia commons.
> 
> can you give an example?

Of course, it's easy to find many examples.

I ran a query in my hometown for PoCs that have wikimedia commons but no wikidata linked.  It's here: https://overpass-turbo.eu/s/1uC9

I then selected one at random: https://www.openstreetmap.org/node/2532890476

It's a nice statue in Kadıköy: https://commons.wikimedia.org/wiki/File:VecihiH%C3%BCrku%C5%9FStatueKad%C4%B1k%C3%B6y.JPG

It has an image in wikimedia commons but is not prominent enough to have a wikidata entry.

**Comment by 1ec5** [comment]
That definitely is [notable enough for Wikidata](https://www.wikidata.org/wiki/Wikidata:Notability), even if it isn’t notable enough for Wikipedia. You can make the item fulfill the “structural need” criterion by adding structured data to the image that references the item: go to the Structured Data tab on the file description page, then add the item in a [depicts](https://www.wikidata.org/wiki/Property:P180) statement. The file currently says it depicts [Sabiha Gökçen](https://www.wikidata.org/wiki/Q459104), but it’s more accurate to say that it depicts the statue depicting Sabiha Gökçen.

The Wikidata and Commons communities might not appreciate it if you mass-added a photo and item about each individual cobblestone in town, but as long as you exercise the same discretion you exercise in uploading to Commons, it should be fine.

**Comment by ddpasa** [comment]
> That definitely is [notable enough for Wikidata](https://www.wikidata.org/wiki/Wikidata:Notability), even if it isn’t notable enough for Wikipedia. You can make the item fulfill the “structural need” criterion by adding structured data to the image that references the item: go to the Structured Data tab on the file description page, then add the item in a [depicts](https://www.wikidata.org/wiki/Property:P180) statement. The file currently says it depicts [Sabiha Gökçen](https://www.wikidata.org/wiki/Q459104), but it’s more accurate to say that it depicts the statue depicting Sabiha Gökçen.
> 
> The Wikidata and Commons communities might not appreciate it if you mass-added a photo and item about each individual cobblestone in town, but as long as you exercise the same discretion you exercise in uploading to Commons, it should be fine.

I don't quite understand the argument you are making.  There is no guarantee that anything on wikimedia commons will also be on wikidata.   By enforcing this arbitrary constraint that everything has to come from wikidata, you are excluding from openstreetmaps of useful (and increasingly popular) tags.

**Comment by 1ec5** [comment]
> There is no guarantee that anything on wikimedia commons will also be on wikidata.

You’re right, but my point is that Wikidata’s notability criteria are so straightforward for real-world features to satisfy – and automated edits are so welcomed on Wikidata – that creating Wikidata items could be a viable alternative to ensuring comprehensive coverage of `wikimedia_commons` tags in OSM through iD.

> By enforcing this arbitrary constraint that everything has to come from wikidata, you are excluding from openstreetmaps of useful (and increasingly popular) tags.

To be clear, iD is not stripping out `wikimedia_commons` tags. It even has a dedicated field for these tags. So there are no constraints to speak of.

If I understand correctly, your proposal is to _automatically_ add the tag. Automatically adding `wikidata` based on `wikipedia` was already controversial at one point (somewhere in [these discussions](https://wiki.openstreetmap.org/wiki/User:Minh_Nguyen/Wikidata_discussions)); adding `wikimedia_commons` based on `wikidata` might need a pretty strong justification to overcome skepticism of Wikidata.

**Comment by ddpasa** [comment]
> > There is no guarantee that anything on wikimedia commons will also be on wikidata.
> 
> You’re right, but my point is that Wikidata’s notability criteria are so straightforward for real-world features to satisfy – and automated edits are so welcomed on Wikidata – that creating Wikidata items could be a viable alternative to ensuring comprehensive coverage of `wikimedia_commons` tags in OSM through iD.
> 
> > By enforcing this arbitrary constraint that everything has to come from wikidata, you are excluding from openstreetmaps of useful (and increasingly popular) tags.
> 
> To be clear, iD is not stripping out `wikimedia_commons` tags. It even has a dedicated field for these tags. So there are no constraints to speak of.
> 
> If I understand correctly, your proposal is to _automatically_ add the tag. Automatically adding `wikidata` based on `wikipedia` was already controversial at one point (somewhere in [these discussions](https://wiki.openstreetmap.org/wiki/User:Minh_Nguyen/Wikidata_discussions)); adding `wikimedia_commons` based on `wikidata` might need a pretty strong justification to overcome skepticism of Wikidata.

If I understand you correctly, you are saying that openstreetmaps contributors should stop adding wikimedia_commons tags to the maps, but instead add the same items to the wikidata (and then link wikidata instead).  I'm not aware of any community decision in this direction.  In fact, most wikimedia tags in openstreetmaps do not have an accompanying wikidata tag (39% have wikidata, 61% don't), as seen in taginfo https://taginfo.openstreetmap.org/keys/wikimedia_commons#combinations.

I don't quite get what the controversy is.  wikidata and wikimedia_commons are separate tags, that sometimes appear together (currenly about 39% of the time).  When they do appear together, ID should make it easier for the mapper and add both.

**Comment by 1ec5** [comment]
> If I understand you correctly, you are saying that openstreetmaps contributors should stop adding wikimedia_commons tags to the maps, but instead add the same items to the wikidata (and then link wikidata instead).

That is not what I’m saying. It’s one thing for mappers to manually add a `wikimedia_commons` tag using either the existing Wikimedia Commons Page field or the raw tag editor; it’s another matter for iD to add it automatically. I was only pointing out that the community would need to be comfortable with an editor adding this key in potentially large numbers.

> sometimes appear together (currenly about 39% of the time)

In fact, according to that link, less than 2% of features with `wikidata` have `wikimedia_commons`, but this number could rise with the proposed feature.

**Comment by tyrasd** [comment]
As mentioned above: this would not add any significant benefit in practice.

**Comment by ddpasa** [comment]
> As mentioned above: this would not add any significant benefit in practice.

While I am okay with closing this as 'not planned', I think it is unfair to say 'would not add any significant benefit in practice'.  It would add a lot of value.
